t/run_after_spec.t

```diff
@@ -0,0 +1,57 @@
+use Test::Spec;
+my($first_nested_after, $second_nested_after, $inner_after, $second_inner_after, $todo_after);
+
+describe "outer describe block" => sub {
+    describe "inner block" => sub {
+        describe "first nested block" => sub {
+            it "runs first" => sub {
+                ok(! $first_nested_after ,"first nested block");
+            };
+            after all => sub {
+                $first_nested_after = 1;
+                ok(1, "after all - first nested block");
+            };
+        };
+        describe "second nested block" => sub {
+            it "first nested after has run by now" => sub {
+                ok($first_nested_after, "first nested after has run");
+            };
+            it "doesn't run second nest after until all it's have run" => sub {
+                ok($second_nested_after, "second nested after hasn't run yet")
```

Comment:
Cool!

Cheers.

On Wed, Jun 1, 2016 at 4:33 AM, Andy Jones <notifications@github.com> wrote:

> In t/run_after_spec.t
> <https://github.com/kingpong/perl-Test-Spec/pull/33#discussion_r65321629>:
>
> > +    describe "inner block" => sub {
> > +        describe "first nested block" => sub {
> > +            it "runs first" => sub {
> > +                ok(! $first_nested_after ,"first nested block");
> > +            };
> > +            after all => sub {
> > +                $first_nested_after = 1;
> > +                ok(1, "after all - first nested block");
> > +            };
> > +        };
> > +        describe "second nested block" => sub {
> > +            it "first nested after has run by now" => sub {
> > +                ok($first_nested_after, "first nested after has run");
> > +            };
> > +            it "doesn't run second nest after until all it's have run" => sub {
> > +                ok($second_nested_after, "second nested after hasn't run yet")
>
> lovely, thanks! I'll try to get to this in the next couple of days (sorry,
> working late most of this week)
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/kingpong/perl-Test-Spec/pull/33/files/bff0254e79b3bf5c14c708213498be62ab5d2dda#r65321629>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AC3QAznb4Kt5yG4Rh6S-1yVOITZCY6kVks5qHUPkgaJpZM4In4PI>
> .
>